### ISSUE TITLE
[Spike] Make cert-manager optional

### ISSUE THREAD

**Issue by jimmykarily** [epinio/epinio#1192]
Part of: https://github.com/epinio/epinio/discussions/1190

Note: UI might need a certificate and maybe there is no way around it (cc @richard-cox )

We want to be able to install an use Epinio even when cert-manager if not installed. We have 2 options:

- Detect cert manager installation by checking for the relevant CRDs. If it's not there, create Ingresses without TLS. That includes Epinio API server, internal registry and applications
- Create the certificates without cert manager (using some pre-deploy helm hook Job?). Can we do the same for apps? If not, maybe do detection of cert-manager for apps but create certs for Epinio API/registry/UI in some other way?

Let's try out the above and see what is possible.

**Comment by jimmykarily** [comment]
We discussed that with @enrichman and @manno and we decided to make cert-manager a hard dependency for Epinio, which should be installed by the user. The reasons are the following:

- cert-manager is easy to install (just a helm command), maybe even already there
- Some parts in Epinio, require an https connection. For example, buildpacks [will refuse to push to a registry using plain http](https://github.com/buildpacks/tekton-integration/issues/31#issuecomment-863290558). We can solve this by allowing the user, during installation, to specify existing certs/secrets. The installation becomes a bit more complex this way. Also, at runtime we won't be able to [create certs for apps](https://github.com/epinio/epinio/blob/68476281362a7358d4a89cf70d76b7dad0ca3b6c/internal/application/ingresses.go#L158) unless the user gives us an existing secret for each of the routes mapped to this app. Again, this requires changes in how routes are specified (we will need to ask for the secret too). So in general, bring your own certs is a complex story with complex flows. I'll let @agracey decide if we want to support that, given how easy it works with cert-manager.

Making the cert-manager a sub-chart of the Epinio chart won't work because we need to install cert-manager first, then wait until it's ready before we create any `Certificate` resources. That's the very reason why we created the epinio-installer in the first place (e.g. read this: https://github.com/helm/helm/issues/8560).

We suggest that we just remove cert-manager from the installer chart and we document that it's a dependency that has to be installed before Epinio. Soon, we will get rid of the installer chart altogether so this documentation should be in the epinio chart README.
